lib/spack/spack/ci.py

```diff
@@ -1978,8 +1979,9 @@ def process_command(cmd, cmd_args, repro_dir):
     # Run the generated install.sh shell script as if it were being run in
     # a login shell.
     try:
+        # Note timeout on wait() should be safe provided stdout, stderr != PIPE
         cmd_process  = subprocess.Popen(['bash', '-l', './{0}'.format(script)])
-        cmd_process.wait()
+        cmd_process.wait(timeout=timeout)
```

Comment:
I have been reading docs for this lately (to code the attempt in #31873) and I think the `timeout` argument is supported only in Python 3 :cry:

Reply:
Following up, this was discussed offline and it was decided adding a timeout for tests was not the way to go.